src/Products/PageTemplates/engine.py

```diff
@@ -215,7 +214,26 @@ def beginScope(self, *args, **kw):
 
     # delegate all else to ``__z_context``
     def __getattr__(self, attr):
-        return getattr(self.__z_context, attr)
+        zc = self.__z_context
+        a = getattr(zc, attr)
+        if getattr(a, "__self__", None) is zc:
+            # a method - we must rebind ``a`` to
+            # ensure it accesses the right ``vars``
+            a = type(a)(a.__func__, self)
+        return a
+
+    # put attributes into ``__z_context``
+    def __setattr__(self, attr, v):
+        setattr(self.__z_context, attr, v)
+
+    def __delattr__(self, attr):
+        delattr(self.__z_context, attr)
+
+    # auxiliary
+    def _setattr(self, attr, v):
+        # currently used only for *attr* with ``__`` prefix
+        super(_C2ZContextWrapper, self).__setattr__(
```

Comment:
You can shorten `super(_C2ZContextWrapper, self)`  to just `super()` and get rid of the line break

Reply:
This works in Python 3 but not in Python 2:
`TypeError: super() takes at least 1 argument (0 given)`

Reply:
You're right, sorry for the noise